src/lib/y2security/clients/security_policy_proposal.rb

```diff
@@ -128,17 +128,30 @@ def links
         links.flatten.compact.uniq
       end
 
+      # Returns the warning message
+      #
+      # @return [String,nil] warning message or nil if there are no failing rules
+      def warning
+        return nil if success?
+
+        _("The current configuration does not comply with the enabled security policies.")
+      end
+
+      # Whether the proposal was successful
+      #
+      # @return [Boolean] true if the proposal was successful; false otherwise
+      def success?
+        rules = failing_rules.values.flatten
+        rules.none? || rules.none?(&:enabled?)
+      end
+
       # Returns the warning level
       #
       # Blocker if there are enabled failing rules
       #
       # @return [Symbol] :blocker
       def warning_level
-        rules = failing_rules.values.flatten
-
-        return :warning if rules.none? || rules.none?(&:enabled?)
-
-        :blocker
+        success? ? :warning : :blocker
```

Comment:
From [ProposalClient#make_proposal](https://www.rubydoc.info/github/yast/yast-yast2/master/Installation/ProposalClient#make_proposal-instance_method) I understand that `warning_level` is only relevant if `warning` is not `nil`

```suggestion
        success? ? nil : :blocker
```